ophyd/controls/signal.py

```diff
@@ -423,20 +450,25 @@ def describe(self):
         dict
             Dictionary of name and formatted description string
         """
-        return {self.name: {'source': 'PV:{}'.format(self._read_pv.pvname),
-                            'dtype': 'number',
+        return {self.name: {'source': 'PV:%s' % self._read_pv.pvname,
+                            'dtype': self.dtype,
+                            'formatter': self.formatter,
                             'shape': []}}
 
     def read(self):
-        """Read the signal and format for data collection
+        """Read the signal and combine it with its timestamp.
+
+        This value is formatted according to the values in self.dtype and
+        self.format
 
         Returns
         -------
         dict
             Dictionary of value timestamp pairs
+            {'value': value, 'timestamp': timestamp}
         """
-
-        return {self.name: {'value': self.value,
+        value = self.describe()['dtype'](self.formatter % self.value)
```

Comment:
If your the encoder on your motor is only reliable to .001 degrees (or pick some other unit, it doesn't matter), why in the world would you store the value as 4.736472928464 degrees instead of 4.737 degrees?